### PR TITLE
[feat][elasticsearch-sink] Option to output canonical JSON

### PR DESCRIPTION
**Pull request by nicoloboschi**
Basic implementation of canonical output for the key.
If the key schema is avro or json, we sort the fields alphabetically (and recursively).

Implementation notes:
- For JSON, we have to parse and rewrite the entire JSON
- For avro, in this pull, the implementation efficiency can be improved. At the moment: Avro record -> unordered JSON -> ordered JSON. We can remove the intermediate json

### CODE REVIEW

**Review by eolivelli — CHANGES_REQUESTED** [review]
overall LGTM

I left one comment about the test

File: pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchExtractTests.java [hunk 1]
```diff
@@ -286,4 +287,144 @@ public Object getNativeObject() {
...
+                    "compatibilityMode", "ELASTICSEARCH",
+                    "schemaEnable", "true",
+                    "canonicalKeyFields", "true",
+                    "keyIgnore", "false"), null);
+            pair = elasticSearchSink.extractIdAndDocument(genericObjectRecord);
+            assertEquals(pair.getKey(), "[\"a_key\",\"0b_key\",\"c_key\",{\"a_inside_inner\":\"a_value_from_inner\",\"b_inside_inner\":\"0b_value_from_inner\"}]");
+        }
+
+        {
+            RecordSchemaBuilder keySchemaBuilder = org.apache.pulsar.client.api.schema.SchemaBuilder.record("key");
```
Comment (cbornet) [inline pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchExtractTests.java:344]: Can this go to a distinct test with explicit naming explaining the difference of what we are testing.
Also not a big fan of scoping inside functions.

**Comment by nicoloboschi** [comment]
CI is ok. If you open the job everything passed (except for a couple of flakies and owasp)

**Review by eolivelli — APPROVED** [review]

**Review by lhotari — COMMENTED** [review]
Good work!

**Pull request merged**